src/xdocs/config_regexp.xml

```diff
@@ -554,6 +554,134 @@
       </subsection>
     </section>
 
+    <section name="RegexpOnFilename">
+      <subsection name="Description">
+        <p>
+          A check that makes sure that files that match specific patterns are located in their proper folders and don't show up in the wrong ones.
+          It can also be used to verify files match specific naming patterns not covered by other checks (Ex: properties, xml, etc.).
+        </p>
+      </subsection>
+
+      <subsection name="Properties">
+        <table>
+          <tr>
+            <th>name</th>
+            <th>description</th>
+            <th>type</th>
+            <th>default value</th>
+          </tr>
+          <tr>
+            <td>folderPattern</td>
+            <td>Regular expression to match the folder path against.</td>
+            <td><a href="property_types.html#regexp">regular expression</a></td>
+            <td><code>null</code></td>
+          </tr>
+          <tr>
+            <td>fileNamePattern</td>
+            <td>Regular expression to match the file path against.</td>
+            <td><a href="property_types.html#regexp">regular expression</a></td>
+            <td><code>null</code></td>
+          </tr>
+          <tr>
+            <td>match</td>
+            <td>Whether to look for a match or mis-match on the file name.</td>
+            <td><a href="property_types.html#boolean">Boolean</a></td>
+            <td><code>true</code></td>
+          </tr>
+          <tr>
+            <td>ignoreFileExtensions</td>
+            <td>Whether to ignore the file extension for the file name match.</td>
+            <td><a href="property_types.html#boolean">Boolean</a></td>
+            <td><code>false</code></td>
+          </tr>
+          <tr>
+            <td>fileExtensions</td>
+            <td>file type extension of files to process</td>
+            <td><a href="property_types.html#stringSet">String Set</a></td>
+            <td><code>{}</code></td>
+          </tr>
+        </table>
+      </subsection>
+
+      <subsection name="Examples">
+        <p>
+          To configure the check to report file names that contain a space:
+        </p>
+        <source>
+&lt;module name=&quot;RegexpOnFilename&quot;/&gt;
+        </source>
+        <p>
+          To configure the check to force picture files to not be 'gif':
+        </p>
+        <source>
+&lt;module name=&quot;RegexpOnFilename&quot;&gt;
+  &lt;property name=&quot;fileNamePattern&quot; value=&quot;\\.gif$&quot;/&gt;
+&lt;/module&gt;
+        </source>
+        <p>
+          OR:
+        </p>
+        <source>
+&lt;module name=&quot;RegexpOnFilename&quot;&gt;
+  &lt;property name=&quot;fileNamePattern&quot; value=&quot;.&quot;/&gt;
+  &lt;property name=&quot;fileExtensions&quot; value=&quot;gif$&quot;/&gt;
+&lt;/module&gt;
+        </source>
+        <p>
+          To configure the check to only allow java or xml files in your folders:<br />
```

Comment:
@rnveach Instead of `To configure the check to only allow java or xml files in your folders:` let's use `To configure the check to only allow Java or XML files in your folders use the below.`, because there's also note before the example.